### PR TITLE
openapi: Add display attributes for token/

### PR DESCRIPTION
**Pull request by averche**
Please see https://github.com/hashicorp/vault/pull/19319 for more details on how this will affect the generated OpenAPI schema.

____

### The following OperationID's will be generated for `token/`:

token-list-accessors
token-create
token-create-orphan
token-create-against-role
token-look-up
token-look-up2
token-look-up-accessor
token-look-up-self
token-look-up-self2
token-renew
token-renew-accessor
token-renew-self
token-revoke
token-revoke-accessor
token-revoke-orphan
token-revoke-self
token-list-roles
token-read-role
token-write-role
token-delete-role
token-tidy

### CODE REVIEW

File: vault/token_store.go [hunk 1]
```diff
@@ -236,6 +254,12 @@ func (ts *TokenStore) paths() []*framework.Path {
 		{
 			Pattern: "create/" + framework.GenericNameRegex("role_name"),
 
+			DisplayAttrs: &framework.DisplayAttributes{
+				OperationPrefix: operationPrefixToken,
+				OperationVerb:   "create",
+				OperationSuffix: "against-role",
```
Comment (AnPucel) [inline vault/token_store.go:260]: should this be `with-role`

File: vault/token_store.go [hunk 2]
```diff
@@ -382,16 +411,30 @@ func (ts *TokenStore) paths() []*framework.Path {
 		{
 			Pattern: "lookup",
 
+			DisplayAttrs: &framework.DisplayAttributes{
+				OperationPrefix: operationPrefixToken,
+			},
```
Comment (AnPucel) [inline vault/token_store.go:416]: does the suffix need to be `lookup` here?

File: vault/token_store.go [hunk 3]
```diff
@@ -382,16 +411,30 @@ func (ts *TokenStore) paths() []*framework.Path {
...
+				logical.ReadOperation: &framework.PathOperation{
+					Callback: ts.handleLookup,
+					DisplayAttrs: &framework.DisplayAttributes{
+						OperationVerb: "lookup",
+					},
+				},
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback: ts.handleLookup,
+					DisplayAttrs: &framework.DisplayAttributes{
+						OperationVerb: "lookup2",
```
Comment (AnPucel) [inline vault/token_store.go:435]: what does `lookup2` mean? Shouldn't it be differentiated by the type of operation i.e. `Update` vs `Read`

File: vault/token_store.go [hunk 4]
```diff
@@ -419,16 +468,31 @@ func (ts *TokenStore) paths() []*framework.Path {
 		{
 			Pattern: "lookup-self$",
 
+			DisplayAttrs: &framework.DisplayAttributes{
+				OperationPrefix: operationPrefixToken,
+				OperationVerb:   "lookup",
```
Comment (AnPucel) [inline vault/token_store.go:473]: should this be `lookup-self` to differentiate between `lookup`

File: vault/token_store.go (hunk 1 above)
Comment (averche) [reply vault/token_store.go:260]: I was basing this on the name of the request handler:

https://github.com/hashicorp/vault/blob/39ab2efb2d276fd5cb2af56697f8256ccfa4aba2/vault/token_store.go#L327-L329

File: vault/token_store.go (hunk 2 above)
Comment (averche) [reply vault/token_store.go:416]: The display attributes are specified under the Operations below

File: vault/token_store.go (hunk 3 above)
Comment (averche) [reply vault/token_store.go:435]: Both Update and Read result in exactly the same operation (`handleLookup`). However, we can't have duplicate operation id's, so these need to be deduped. In such cases, I've been using a "2" suffix for lack of a better name. In the libraries we can filter these out with handlebar templates.

```
token-lookup
token-lookup2
```

File: vault/token_store.go (hunk 4 above)
Comment (averche) [reply vault/token_store.go:473]: There are two suffixes for individual operations below (self/self2)

File: vault/token_store.go [hunk 5]
```diff
@@ -382,16 +411,30 @@ func (ts *TokenStore) paths() []*framework.Path {
...
+				logical.ReadOperation: &framework.PathOperation{
+					Callback: ts.handleLookup,
+					DisplayAttrs: &framework.DisplayAttributes{
+						OperationVerb: "look-up",
+					},
+				},
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback: ts.handleLookup,
+					DisplayAttrs: &framework.DisplayAttributes{
+						OperationVerb: "look-up2",
```
Comment (lursu) [inline vault/token_store.go:435]: Is this the same situation as the `"roles2"` case earlier?

File: vault/token_store.go [hunk 6]
```diff
@@ -419,16 +468,31 @@ func (ts *TokenStore) paths() []*framework.Path {
...
+				logical.ReadOperation: &framework.PathOperation{
+					Callback: ts.handleLookupSelf,
+					DisplayAttrs: &framework.DisplayAttributes{
+						OperationSuffix: "self",
+					},
+				},
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback: ts.handleLookupSelf,
+					DisplayAttrs: &framework.DisplayAttributes{
+						OperationSuffix: "self2",
```
Comment (lursu) [inline vault/token_store.go:493]: same here.

**Review by lursu — APPROVED** [review]

File: vault/token_store.go (hunk 5 above)
Comment (averche) [reply vault/token_store.go:435]: Yes, pretty much, we need to disambiguate here to prevent collisions in OperationID's

**Pull request merged**